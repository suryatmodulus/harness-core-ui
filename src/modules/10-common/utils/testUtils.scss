/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
