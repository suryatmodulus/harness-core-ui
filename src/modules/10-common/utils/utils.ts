/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
