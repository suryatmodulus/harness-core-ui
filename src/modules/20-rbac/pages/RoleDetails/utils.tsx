/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import produce from 'immer'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { Permission, PermissionResponse } from 'services/rbac'

export const getPermissionMap = (permissionList?: PermissionResponse[]): Map<ResourceType, Permission[]> => {
  const permissionsMap: Map<ResourceType, Permission[]> = new Map()
  permissionList?.map(response => {
    if (response.permission.resourceType) {
      if (permissionsMap.has(response.permission.resourceType as ResourceType)) {
        permissionsMap.get(response.permission.resourceType as ResourceType)?.push(response.permission)
      } else permissionsMap.set(response.permission.resourceType as ResourceType, [response.permission])
    }
  })
  return permissionsMap
}

export const onPermissionChange = (
  permission: string,
  isAdd: boolean,
  permissions: string[],
  setPermissions: (value: React.SetStateAction<string[]>) => void
): void => {
  if (isAdd) {
    if (
      permission === PermissionIdentifier.EDIT_DASHBOARD ||
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    ) {
      setPermissions(_permissions => [...permissions, permission, PermissionIdentifier.VIEW_DASHBOARD])
    } else {
      setPermissions(_permissions => [...permissions, permission])
    }
  } else if (
    !(
      permission === PermissionIdentifier.VIEW_DASHBOARD &&
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    )
  ) {
    setPermissions(_permissions =>
      produce(_permissions, draft => {
        draft?.splice(permissions.indexOf(permission), 1)
      })
    )
  }
}
