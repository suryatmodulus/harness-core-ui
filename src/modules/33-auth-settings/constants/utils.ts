/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum AuthenticationMechanisms {
  SAML = 'SAML',
  OAUTH = 'OAUTH',
  LDAP = 'LDAP',
  USER_PASSWORD = 'USER_PASSWORD'
}

export const getSamlEndpoint = (accountId: string): string => {
  let url = window.location.href.split('/#')[0]?.replace('/ng', '')

  if (window.apiUrl) {
    if (window.apiUrl.startsWith('/')) {
      url = `${url}${window.apiUrl}`
    } else {
      url = window.apiUrl
    }
  }

  return `${url}/api/users/saml-login?accountId=${accountId}`
}
