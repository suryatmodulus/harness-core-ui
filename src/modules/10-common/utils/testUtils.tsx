/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { queryByAttribute } from '@testing-library/react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation, useHistory } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uicore'
import qs from 'qs'

import { enableMapSet } from 'immer'
import { AppStoreContext, AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import { LicenseStoreContext, LicenseStoreContextProps } from 'framework/LicenseStore/LicenseStoreContext'
import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
import { withAccountId, accountPathProps } from '@common/utils/routeUtils'
import type { Project } from 'services/cd-ng'
import { StringsContext } from 'framework/strings'

import { FeaturesContext, FeaturesContextProps } from 'framework/featureStore/FeaturesContext'
import type { FeatureDetail, FeatureMetaData } from 'framework/featureStore/featureStoreUtil'
import type { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import './testUtils.scss'
import { PermissionsContext, PermissionsContextProps } from 'framework/rbac/PermissionsContext'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseGetMockDataWithMutateAndRefetch<T> extends UseGetMockData<T> {
  mutate: () => Record<string, unknown>
  refetch: () => Record<string, unknown>
  cancel?: () => Record<string, unknown>
}

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): HTMLElement | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): HTMLElement | null => document.querySelector('.bp3-popover-content')

export interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, unknown>
  defaultAppStoreValues?: Partial<AppStoreContextProps>
  defaultLicenseStoreValues?: Partial<LicenseStoreContextProps>
  defaultPermissionValues?: Partial<PermissionsContextProps>
  defaultFeaturesValues?: Partial<FeaturesContextProps>
  projects?: Project[]
  enableBrowserView?: boolean
  stringsData?: Record<string, string>
  getString?(key: string): string
}

export const prependAccountPath = (path: string): string => withAccountId(() => path)(accountPathProps)

export const CurrentLocation = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export interface BrowserViewProps {
  enable?: boolean
  children: React.ReactNode
}

export function BrowserView(props: BrowserViewProps): React.ReactElement {
  const { enable, children } = props
  const location = useLocation()
  const history = useHistory()

  if (!enable) {
    return <>{children}</>
  }

  function handlePathChange(e: React.ChangeEvent<HTMLInputElement>): void {
    history.replace(e.currentTarget.value)
  }

  const search = location.search ? `?${location.search.replace(/^\?/, '')}` : ''

  return (
    <div className="browser">
      <div className="browser-header">
        <input className="browser-path" value={location.pathname + search} onChange={handlePathChange} />
      </div>
      <div className="browser-content">{children}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  enableMapSet()
  const {
    path = '/',
    pathParams = {},
    defaultAppStoreValues,
    queryParams = {},
    defaultLicenseStoreValues,
    defaultPermissionValues,
    defaultFeaturesValues,
    stringsData = {},
    getString = (key: string) => key
  } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const history = React.useMemo(() => createMemoryHistory({ initialEntries: [routePath] }), [])

  const defaultReturn = {
    enabled: true
  }

  /** TODO: Try fixing this later. This is causing some tests to fail */
  // React.useEffect(() => {
  //   history.replace(compile(path)(pathParams) + qs.stringify(queryParams, { addQueryPrefix: true }))
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, [path, pathParams, queryParams])

  return (
    <StringsContext.Provider value={{ data: stringsData as any, getString }}>
      <AppStoreContext.Provider
        value={{
          featureFlags: {
            FEATURE_ENFORCEMENT_ENABLED: true
          },
          updateAppStore: () => void 0,
          currentUserInfo: {},
          ...defaultAppStoreValues
        }}
      >
        <LicenseStoreContext.Provider
          value={{
            versionMap: {},
            licenseInformation: {},
            CI_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            FF_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CCM_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CD_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            updateLicenseStore: () => void 0,
            ...defaultLicenseStoreValues
          }}
        >
          <PermissionsContext.Provider
            value={{
              permissions: new Map<string, boolean>(),
              requestPermission: () => void 0,
              checkPermission: () => true,
              cancelRequest: () => void 0,
              ...defaultPermissionValues
            }}
          >
            <FeaturesContext.Provider
              value={{
                features: new Map<FeatureIdentifier, FeatureDetail>(),
                featureMap: new Map<FeatureIdentifier, FeatureMetaData>(),
                getEdition: () => {
                  return undefined
                },
                requestFeatures: () => void 0,
                requestLimitFeature: () => void 0,
                checkFeature: () => {
                  return defaultReturn
                },
                checkLimitFeature: () => {
                  return defaultReturn
                },
                getRestrictionType: () => {
                  return undefined
                },
                ...defaultFeaturesValues
              }}
            >
              <Router history={history}>
                <ModalProvider>
                  <RestfulProvider base="/">
                    <BrowserView enable={props.enableBrowserView}>
                      <Switch>
                        <Route exact path={path}>
                          {props.children}
                        </Route>
                        <Route>
                          <CurrentLocation />
                        </Route>
                      </Switch>
                    </BrowserView>
                  </RestfulProvider>
                </ModalProvider>
              </Router>
            </FeaturesContext.Provider>
          </PermissionsContext.Provider>
        </LicenseStoreContext.Provider>
      </AppStoreContext.Provider>
    </StringsContext.Provider>
  )
}

export const queryByNameAttribute = (name: string, container: HTMLElement): HTMLElement | null =>
  queryByAttribute('name', container, name)
