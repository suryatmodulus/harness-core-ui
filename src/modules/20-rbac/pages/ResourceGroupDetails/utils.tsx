import produce from 'immer'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector } from '@rbac/utils/utils'
import type { ResourceSelector } from 'services/resourcegroups'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (isDynamicResourceSelector(resourceSelector.type)) {
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    } else {
      map.set(resourceSelector.resourceType, resourceSelector.identifiers)
    }
  })
  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}
