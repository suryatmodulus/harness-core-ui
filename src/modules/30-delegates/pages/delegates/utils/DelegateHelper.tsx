/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useStrings } from 'framework/strings'
import { DelegateTypes } from '@delegates/constants'

export const GetDelegateTitleTextByType = (type: string): string => {
  const { getString } = useStrings()

  switch (type) {
    case DelegateTypes.KUBERNETES_CLUSTER:
      return getString('kubernetesText')
    default:
      /* istanbul ignore next */
      return ''
  }
}
