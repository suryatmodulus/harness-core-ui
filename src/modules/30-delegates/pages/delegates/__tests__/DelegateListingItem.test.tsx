/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, act, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { TestWrapper } from '@common/utils/testUtils'
import DelegatesListingItem, { DelegateListingHeader } from '../DelegateListingItem'
import { delegateGroupsMock } from './DelegateGroupsMock'

jest.mock('@common/components/YAMLBuilder/YamlBuilder')

const setOpenTroubleshoterFn = jest.fn()

jest.mock('services/portal', () => ({
  useDeleteDelegateGroupByIdentifier: () => ({
    mutate: jest.fn()
  })
}))

describe('DelegateListingHeader test', () => {
  test('render DelegateListingHeader', () => {
    const { container } = render(
      <TestWrapper>
        <DelegateListingHeader />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('Delegates Listing With Groups', () => {
  test('render data', () => {
    const { container } = render(
      <TestWrapper>
        <DelegatesListingItem delegate={delegateGroupsMock[0]} setOpenTroubleshoter={setOpenTroubleshoterFn} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
  test('Click on item', async () => {
    const { container } = render(
      <TestWrapper>
        <DelegatesListingItem delegate={delegateGroupsMock[0]} setOpenTroubleshoter={setOpenTroubleshoterFn} />
      </TestWrapper>
    )
    act(() => {
      fireEvent.click(container.firstChild!)
    })
    expect(container).toMatchSnapshot()
  })
  test('click on delegate item action', async () => {
    const { getAllByText, container } = render(
      <TestWrapper>
        <DelegatesListingItem delegate={delegateGroupsMock[1]} setOpenTroubleshoter={setOpenTroubleshoterFn} />
      </TestWrapper>
    )
    const menuBtn = container.querySelector('button') as HTMLButtonElement
    userEvent.click(menuBtn!)
    userEvent.click(getAllByText('delegates.troubleshootOption')[0]!)
    expect(document.body.innerHTML).toContain('troubleshoot')
  })
  test('click on delegate item delete action', async () => {
    const { getAllByText, container, queryAllByText } = render(
      <TestWrapper>
        <DelegatesListingItem delegate={delegateGroupsMock[1]} setOpenTroubleshoter={setOpenTroubleshoterFn} />
      </TestWrapper>
    )
    const menuBtn = container.querySelector('button') as HTMLButtonElement
    act(() => {
      fireEvent.click(menuBtn!)
    })
    await waitFor(() => {
      expect(getAllByText('delete')[0]).toBeDefined()
    })
    const deleteBtn = getAllByText('delete')[0]
    act(() => {
      fireEvent.click(deleteBtn!)
    })
    expect(document.body.querySelector('[class*="useConfirmationDialog"]')).toBeDefined()
    const modalDeleteBtn = queryAllByText('delete')[1]
    act(() => {
      fireEvent.click(modalDeleteBtn!)
    })
    waitFor(() => {
      expect(document.body.innerHTML).not.toContain('useConfirmationDialog')
    })
  })
})
