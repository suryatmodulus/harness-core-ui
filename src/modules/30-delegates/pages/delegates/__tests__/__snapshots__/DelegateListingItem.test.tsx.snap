// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DelegateListingHeader test render DelegateListingHeader 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main delegateListHeader"
  >
    <div
      style="width: 80px;"
    />
    <div
      style="width: 25%;"
    >
      delegate.DelegateName
    </div>
    <div
      style="width: 20%;"
    >
      tagsLabel
    </div>
    <div
      style="width: 20%;"
    >
      instanceFieldOptions.instances
    </div>
    <div
      style="width: calc(15% - 40px);"
    >
      delegate.LastHeartBeat
    </div>
    <div
      style="width: calc(15% - 40px);"
    >
      connectivityStatus
    </div>
    <div
      style="width: 5%;"
    />
  </div>
</div>
`;

exports[`Delegates Listing With Groups Click on item 1`] = `
<div>
  <div>
    <h1>
      Not Found
    </h1>
    <div
      data-testid="location"
    >
      /account/undefined/settings/resources/delegate/delGroupIdentifier1
    </div>
  </div>
</div>
`;

exports[`Delegates Listing With Groups render data 1`] = `
<div>
  <div
    class="bp3-card bp3-interactive bp3-elevation-2 Card--card delegateItemContainer Card--interactive"
    tabindex="0"
  >
    <div
      class="Layout--horizontal StyledProps--main delegateItemSubcontainer"
    >
      <div
        class="delegateItemIcon"
        style="width: 80px;"
      >
        <span
          class="bp3-icon bp3-icon-chevron-right StyledProps--main expandIcon"
          icon="chevron-right"
        >
          <svg
            data-icon="chevron-right"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <desc>
              chevron-right
            </desc>
            <path
              d="M13.71 9.29l-6-6a1.003 1.003 0 00-1.42 1.42l5.3 5.29-5.29 5.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l6-6c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall"
            data-icon="app-kubernetes"
          >
            <svg
              height="24"
              viewBox="0 0 32 30"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
              >
                <path
                  d="M15.94.023c-.228 0-.457 0-.685.229L4.512 5.28c-.457.229-.915.686-1.143 1.372L.626 18.08c-.229.457 0 1.143.229 1.6l.228.229 7.314 9.143c.458.457.915.685 1.6.685h11.886c.686 0 1.143-.228 1.6-.685l7.314-9.143c.458-.457.458-1.143.458-1.6L28.512 6.88c-.229-.685-.457-1.143-1.143-1.371L16.626.48c0-.228-.457-.457-.686-.457z"
                  fill="#326CE5"
                />
                <path
                  d="M15.94 3.909c-.457 0-.685.228-.685.686v2.057c0 .228-.229.228-.229.228v.229c-.457 0-.686 0-1.143.228-1.6.458-3.2 1.143-4.114 2.286l-.229-.228h-.457c-.228-.23-.457-.458-.914-.686l-.457-.457h-.229c-.228 0-.228-.229-.457-.229s-.457 0-.457.229c-.229.228-.229.685.228.914h.23c.228.229.456.229.456.229.457.228.686.457.914.685 0 0 .23.229.23.457l.228.229c-1.143 1.6-1.6 3.657-1.372 5.714h-.228l-.229.229c-.229 0-.686.228-1.143.228H4.97c-.457 0-.686.458-.457.686 0 .229.457.457.914.457h.229c.228 0 .457-.228.685-.228.457-.229.686-.229 1.143-.229.229 0 .229 0 .457.229h.229c.686 2.057 2.057 3.657 3.657 4.571l-.229.229v.457c-.228.228-.228.686-.685 1.143-.229.228-.229.228-.229.457v.228c-.228.458 0 .686.229.915.228.228.685 0 .914-.457v-.23c0-.228.229-.456.229-.685.228-.457.228-.914.457-1.143 0 0 .228 0 .228-.228l.229-.229c1.371.457 3.2.686 4.8.457.457 0 .686-.228 1.143-.228l.228.228s.229 0 .229.229c.228.229.228.686.457 1.143 0 .228 0 .457.229.686v.228c.228.457.457.457.914.457.228-.228.457-.457.228-.914v-.229c0-.228-.228-.228-.228-.457-.229-.457-.457-.685-.457-.914v-.686c1.828-1.143 2.971-2.743 3.657-4.571h.228s.23-.229.458-.229.685.229 1.142.229c.23 0 .458.228.686.228h.229c.457 0 .685-.228.914-.457 0-.228-.229-.685-.457-.685h-.914c-.458 0-.915 0-1.143-.23-.229 0-.229-.228-.229-.228h-.228c.228-.914 0-2.057-.229-2.971-.229-.914-.686-2.057-1.143-2.743l.229-.228c0-.23 0-.23.228-.458.229-.228.457-.457.915-.685.228 0 .457-.229.457-.229h.228c.229-.228.457-.686.229-.914-.229-.229-.686-.229-.914 0h-.229l-.457.457c-.229.229-.686.686-.914.686h-.458l-.228.228a8.778 8.778 0 00-5.257-2.514v-.229l-.229-.228V5.052c-.457-.686-.686-1.143-1.143-1.143zm-.685 5.028l-.229 3.429a.492.492 0 01-.457.457h-.229l-2.743-2.057c.915-.914 1.83-1.371 3.2-1.829h.458zm1.6 0c1.371.229 2.742.915 3.885 1.829l-2.743 1.829c-.228.228-.685.228-.685 0l-.229-.23-.228-3.428zm-6.4 3.2l2.514 2.286c.228.229.228.457 0 .686l-.229.228-3.2.915c-.228-1.6 0-2.972.915-4.115zm11.2 0c.457.686.685 1.372.914 2.058.228.685.228 1.37.228 2.057l-3.2-.915c-.228 0-.457-.457-.457-.685 0-.229 0-.229.229-.229l2.286-2.286zm-6.172 2.286h.914l.686.686-.228.914-.915.457-.914-.457-.229-.914.686-.686zm3.2 2.743c.229 0 .229 0 0 0l3.429.457c-.457 1.372-1.372 2.514-2.743 3.429l-1.372-3.2c.23-.229.23-.457.686-.686h0zm-5.486 0c.23 0 .458.229.458.457v.457l-1.372 3.2c-1.143-.685-2.057-1.828-2.743-3.428l3.657-.686zm2.743 1.371h.229l.228.229 1.6 2.971c-.228 0-.457.229-.685.229-1.143.229-2.515.229-3.657-.229l1.6-2.971c.457 0 .457-.229.685-.229h0z"
                  fill="#FFF"
                  stroke="#FFF"
                  stroke-width="0.25"
                />
              </g>
            </svg>
          </span>
        </p>
      </div>
      <div
        class="Layout--horizontal StyledProps--main"
        data-testid="groupHostName1"
        style="width: 25%;"
      >
        <div
          class="Layout--vertical StyledProps--main"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
            data-testid="delegate-1"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
            >
              delegate-1
            </p>
          </div>
          <div
            class="groupHostName"
          >
            groupHostName1
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main connectivity"
        style="width: 20%;"
      >
        <div
          class="StyledProps--font StyledProps--main"
          style="display: inline-flex; flex-wrap: wrap;"
        >
          <p
            class="StyledProps--font StyledProps--main"
            style="display: inline-block; border-radius: 4px; font-size: 10px;"
          >
            implicitSelector1
          </p>
          <p
            class="StyledProps--font StyledProps--main"
            style="display: inline-block; border-radius: 4px; font-size: 10px;"
          >
            customtag1
          </p>
          <p
            class="StyledProps--font StyledProps--main"
            style="display: inline-block; border-radius: 4px; font-size: 10px;"
          >
            customtag2
          </p>
        </div>
        <span />
      </div>
      <div
        class="Layout--horizontal StyledProps--main instancesColumn"
        style="width: 20%;"
      >
        0
      </div>
      <div
        class="Layout--horizontal StyledProps--main"
        style="width: calc(15% - 40px);"
      >
        dummy date
      </div>
      <div
        class="Layout--vertical StyledProps--main"
        style="width: calc(15% - 40px);"
      >
        <p
          class="StyledProps--font StyledProps--main"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon bp3-icon-full-circle StyledProps--main Text--icon StyledProps--padding-small StyledProps--color StyledProps--color-green600"
            icon="full-circle"
          >
            <svg
              data-icon="full-circle"
              height="6"
              viewBox="0 0 16 16"
              width="6"
            >
              <desc>
                full-circle
              </desc>
              <path
                d="M8 0a8 8 0 100 16A8 8 0 108 0z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          connected
        </p>
      </div>
      <div
        class="Layout--vertical StyledProps--main"
        style="width: 5%;"
      >
        <div
          class="Layout--horizontal StyledProps--main itemActionContainer"
        >
          <span
            class="bp3-popover-wrapper bp3-dark"
          >
            <span
              class="bp3-popover-target"
            >
              <button
                class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                type="button"
              >
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="Options"
                >
                  <svg
                    fill="none"
                    height="16"
                    viewBox="0 0 4 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
